docs/go/streaming.md

```diff
@@ -28,6 +28,13 @@ networking infrastructure and makes your application inaccessible to
 less-sophisticated clients. You can minimize these downsides by keeping
 streams short-lived.
 
+Also, the Go HTTP server timeout applies to the entire operation duration,
+even for streaming calls. So if an operation takes longer than that, the server
+closes the stream and clients can see the error message, which will look
+something like:
+
+`stream error: stream ID 5; INTERNAL_ERROR; received from peer`
```

Comment:
I know this is repeated 3x in here, so making the content more complex makes it a little bit bigger of a maintenance burden for future changes. But it would be nice if this was a little more specific. In this case, the timeouts in question are the `ReadTimeout` and `WriteTimeout` fields of the [`http.Server`](https://pkg.go.dev/net/http#Server). The other timeout fields should not cause this sort of error. In particular, `ReadHeaderTimeout` is not only safe to use (meaning it will not cause this error) but is also strongly encouraged to set on all servers.

Reply:
Maybe that additional info can just be put into the FAQ, and these other places that reiterate the content could be more brief and link to that FAQ entry.